programs/CMakeLists.txt

```diff
@@ -1,8 +1,10 @@
-add_subdirectory(aes)
-add_subdirectory(hash)
+if(!ENABLE_PSA)
+    add_subdirectory(aes)
+    add_subdirectory(hash)
+    add_subdirectory(random)
+endif()
 add_subdirectory(pkey)
-add_subdirectory(random)
-add_subdirectory(ssl)
 add_subdirectory(test)
-add_subdirectory(x509)
 add_subdirectory(util)
+add_subdirectory(ssl)
```

Comment:
If this is alphabetical order, what alphabet are you using?

Reply:
The "put mbedtls specific things at the bottom" alphabet. I've made it actually alphabetical now.